.github/workflows/ci-skip.yml

```diff
@@ -0,0 +1,88 @@
+name: CI skip
+
+on:
+  pull_request:
+    # Any update here needs to be done for the `no-ci-required` step (see below),
+    # and mirrored into `ci.yml`.
+    paths:
+      # Workflows that have no effect on this workflow.
+      - '.github/dependabot.yml'
+      - '.github/workflows/audits.yml'
+      - '.github/workflows/book.yml'
+      - '.github/workflows/ci.yml'
+      - '.github/workflows/lints.yml'
+      - '.github/workflows/release-docker-hub.yml'
+      # Documentation.
+      - 'contrib/debian/copyright'
+      - 'doc/**'
+      - '**.md'
+      - 'COPYING'
+      - 'INSTALL'
+
+jobs:
+  required-pass:
+    name: Bypass required-pass steps
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      # Ensure that the base branch exists.
+      - run: git checkout ${{ github.base_ref }}
+      # Go back to the head ref (not necessary but for consistency).
+      - run: git checkout ${{ github.head_ref }}
+
+      - name: Check whether the changes are only to the set of filtered paths
+        id: no-ci-required
+        run: |
+          cat <<EOF > ./subclass.py
+          import os
+          import re
+          import shlex
+          import sys
+
+          paths = [
+              r'^\.github/dependabot\.yml$',
+              r'^\.github/workflows/audits\.yml$',
+              r'^\.github/workflows/book\.yml$',
+              r'^\.github/workflows/ci\.yml$',
+              r'^\.github/workflows/lints\.yml$',
+              r'^\.github/workflows/release-docker-hub\.yml$',
+              r'^contrib/debian/copyright$',
```

Comment:
nonblocking: what about thinks like the `metrics` contrib? None of that is checked by CI. Shouldn't this filter basically be set up such that anything that CI doesn't check based on is ignored?

Reply:
Indeed; I only looked for the obvious things that didn't require triggering CI when I added that filter in #6836, and this PR is just maintaining that list. We can add further entries to these lists in subsequent PRs (that themselves will cause CI to run because they'll be altering `ci.yml`).